packages/core/src/components/overflow-list/overflowList.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+import classNames from "classnames";
+import * as React from "react";
+import ResizeObserver from "resize-observer-polyfill";
+
+import { Boundary } from "../../common/boundary";
+import * as Classes from "../../common/classes";
+import { IProps } from "../../common/props";
+import { throttle } from "../../common/utils";
+
+export interface IOverflowListProps<T> extends IProps {
+    /**
+     * Which direction the items should collapse from: start or end of the children.
+     * @default Boundary.START
+     */
+    collapseFrom?: Boundary;
+
+    /**
+     * All items to display in the list. Items that don’t fit in the container will be rendered in the overflow instead.
+     */
+    items: T[];
+
+    /**
+     * If `true`, all parent elements of the container will also be observed. If changes to a parent’s size is detected, the overflow will
+     * be recalculated.
+     * Only enable this prop if the overflow should be recalculated when a parent element resizes in a way that does not also cause the
+     * `OverflowContainer` to resize.
+     */
+    observeParents?: boolean;
+
+    /**
+     * Callback invoked to render the overflow if necessary.
+     */
+    overflowRenderer: (items: T[]) => React.ReactNode;
+
+    /**
+     * Callback invoked to render each visible item.
+     */
+    visibleItemRenderer: (item: T, index: number) => React.ReactChild;
+}
+
+export interface IOverflowListState<T> {
+    overflow: T[];
+    visible: T[];
+}
+
+export class OverflowList<T> extends React.Component<IOverflowListProps<T>, IOverflowListState<T>> {
+    public static displayName = "Blueprint2.OverflowList";
+
+    public static defaultProps: Partial<IOverflowListProps<never>> = {
+        collapseFrom: Boundary.START,
+    };
+
+    public static ofType<T>() {
+        return OverflowList as new (props: IOverflowListProps<T>) => OverflowList<T>;
+    }
+
+    private element: Element | null = null;
+    private observer: ResizeObserver;
+    private previousWidths = new Map<Element, number>();
+    private spacer: Element | null = null;
+
+    public constructor(props: IOverflowListProps<T>, context?: any) {
+        super(props, context);
+        this.state = {
+            overflow: [],
+            visible: props.items,
+        };
+
+        this.observer = new ResizeObserver(this.handleResize);
```

Comment:
this constructor is entirely unnecessary. just assign `public state = { overflow: [], visible: this.props.items }` and initialize observer when defining it.